### PR TITLE
LPD-30357 - Update Style Books Import process to handle Token Definition property

### PR DESCRIPTION
**Pull request by algobob**
Re-sent from https://github.com/liferay-platform-experience/liferay-portal/pull/780

Ticket: https://liferay.atlassian.net/browse/LPD-30357

## Motivation
Given the the new Token Definition/Stylebook relationship, the import/export should include the reference to the Frontend Token Definition.

## Proposed Solution
Include **themeId** property when import/export

### CODE REVIEW

**Comment by marcelabc** [comment]
ci:test:sf

File: modules/apps/style-book/style-book-api/src/main/java/com/liferay/style/book/exception/StyleBookEntryThemeIdException.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * @author Anderson Luiz
+ */
+public class StyleBookEntryThemeIdException extends PortalException {
+
+	public static class MustNotBeNull extends StyleBookEntryThemeIdException {
+
+		public MustNotBeNull() {
+			super("Value must not be null");
```
Comment (marcelabc) [inline modules/apps/style-book/style-book-api/src/main/java/com/liferay/style/book/exception/StyleBookEntryThemeIdException.java:18]: When defining the exception message, you should use the property name/column name. Then, you should replace the word `Value` for `Theme ID`.
Can you rewrite it please?

`super("Theme ID must not be null");`

File: modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/ExportImportStyleBookEntriesMVCResourceCommandTest.java [hunk 2]
```diff
@@ -213,11 +220,15 @@ public void testExportImportSingleStyleBookEntryAndOverwrite()
 			ServiceContextTestUtil.getServiceContext(
 				_sourceGroup, TestPropsValues.getUserId());
 
+		String name = RandomTestUtil.randomString();
```
Comment (marcelabc) [inline modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/ExportImportStyleBookEntriesMVCResourceCommandTest.java:226]: You should declare the variable closer to its use.
I see that `name` is used when you create the variable `StyleBookEntry updatedStyleBookEntry =`. Can you please declare `name` closer to its usage?

File: modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/ExportImportStyleBookEntriesMVCResourceCommandTest.java [hunk 3]
```diff
@@ -124,13 +124,15 @@ public void testExportImportSingleStyleBookEntry() throws Exception {
 
 		String name = RandomTestUtil.randomString();
 
+		String themeId = RandomTestUtil.randomString();
+
```
Comment (marcelabc) [inline modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/ExportImportStyleBookEntriesMVCResourceCommandTest.java:128]: Can you sort this variable?
`themeId` should come after `styleBookEntryKey`

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by algobob** [comment]
ci:test:sf

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by ethib137** [comment]
Thanks @marcelabc ! 

@algobob yes if any changes have been made we should always run relevant. 

Since we now have a senior backend approval. I'll forward this PR. Thanks everyone!

**Comment by ethib137** [comment]
ci:forward

**Pull request closed without merging**